bindings/python/tests/test_parse.py

```diff
@@ -0,0 +1,30 @@
+# from context import scripts
+import scripts.parse as parse
+
+import clang.cindex as clang
+
+
+def test_node_in_this_file_true():
+    def recursive_call(cursor):
+        for child in cursor.get_children():
+            assert parse.node_in_this_file(child, tu.spelling) == True
+            recursive_call(cursor=child)
+
+    unsaved_files = (("file.cpp", "void function() {}"),)
+    tu = clang.TranslationUnit.from_source(filename="file.cpp", unsaved_files=unsaved_files)
+    recursive_call(cursor=tu.cursor)
+
+
+def test_node_in_this_file_false_none():
+    def recursive_call(cursor):
+        for child in cursor.get_children():
+            if child.location.file:
```

Comment:
I've seen a limited amount of tests in my developer life, but I've never seen branching assertions. You control the *given* portion of the test, so I'd expect you to know exactly what requirements are necessary for your function to return a set value.

Reply:
The `if child.location.file` branch removed, as `None` case was removed from `parse.py`: https://github.com/divmadan/pcl/pull/6/commits/2e85ac9ba667459664abea4fe3681e7f1abd2ce0